qcfractal/interface/models/task_models.py

```diff
@@ -37,36 +47,83 @@ class BaseResultEnum(str, Enum):
 
 
 class PythonComputeSpec(BaseModel):
-    function: str
-    args: List[Any]
-    kwargs: Dict[str, Any]
+    function: str = Schema(
+        ...,
+        description="The module and function name of a Python-callable to call. Of the form 'module.function'"
+    )
+    args: List[Any] = Schema(
+        ...,
+        description="List of positional arguments to pass into ``function`` in order they appear."
+    )
+    kwargs: Dict[str, Any] = Schema(
+        ...,
+        description="Dictionary of keyword arguments to pass into ``function``."
+    )
 
 
 class TaskRecord(BaseModel):
 
-    id: ObjectId = None
-
-    spec: PythonComputeSpec
-    parser: str
-    status: TaskStatusEnum = "WAITING"
+    id: ObjectId = Schema(
+        None,
+        description="The Database assigned ID of the Task, if it has been assigned yet."
+    )
+
+    spec: PythonComputeSpec = Schema(
+        ...,
+        description="The Python function specification this task will execute or has executed."
```

Comment:
Another good spot to be more declarative. "The Python specification for this task."